File: src/main/kotlin/com/thoughtworks/archguard/system_info/domain/SystemInfo.kt
package com.thoughtworks.archguard.system_info.domain

import java.sql.Timestamp

data class SystemInfo(var id: Long? = null,
                      val systemName: String = "",
                      val repo: String = "",
                      val sql: String = "",
                      val username: String = "",
                      val password: String = "",
                      val scanned: ScannedType = ScannedType.NONE,
                      val qualityGateProfileId: Long? = null,
                      val repoType: String = "GIT",
                      val updatedTime: Timestamp? = null,
                      val badSmellThresholdSuiteId: Long? = null,
                      val branch: String? = "master",
                      val language: String? = "jvm",
                      val codePath: String? = ""
)

File: scan_sourcecode/src/test/resources/languages/ts/apicall/api/addition/systemInfo.ts
import axios from "../axios";
import { baseURL } from "./config";
const systemInfoApi = "/system-info";

export interface SystemInfo {
  id: number;
  systemName: string;
  repo: string[];
  repoType: "GIT" | "SVN";
  username: string;
  password: string;
  sql: string;
  scanned: "NONE" | "SCANNING" | "SCANNED" | "FAILED";
  updatedTime: number;
  badSmellThresholdSuiteId: number;
  branch: string;
}

export function querySystemInfo() {
  return axios<SystemInfo[]>({
    baseURL,
    url: systemInfoApi,
    method: "GET",
  });
}

export function updateSystemInfo(parameter: any) {
  return axios<void>({
    baseURL,
    url: systemInfoApi,
    method: "PUT",
    data: parameter,
  });
}

export function createSystemInfo(parameter: any) {
  return axios<void>({
    baseURL,
    url: systemInfoApi,
    method: "POST",
    data: parameter,
  });
}

export function deleteSystem(systemId: number) {
  return axios<void>({
    baseURL,
    url: `${systemInfoApi}/${systemId}`,
    method: "DELETE",
  });
}
